prototype/KeyGuardMaa/readme.md

```diff
@@ -0,0 +1,125 @@
+﻿# KeyGuard Attestation Proof‑of‑Concept
+
+> **Goal**  Demonstrate how to create a KeyGuard‑protected RSA key on Windows, attest it with Azure **Microsoft Azure Attestation** (MAA), receive a JWT, and prove the key can still perform cryptographic operations.
+
+---
+
+## 1️⃣ Prerequisites
+
+| Requirement | Notes |
+|-------------|-------|
+| Windows **Server 2022** with TPM 2.0 (KeyGuard is Windows‑only; Linux/macOS not supported) | The demo queries TPM logs & AK certs. |
+| KeyGuard with **Virtualisation‑Based Security (VBS)** | Must be enabled (see below) and the **Windows Security Service** running. |
+| .NET 8 SDK + Visual Studio 2022 | Any recent SDK capable of building **C# 12**. |
+| `AttestationClientLib.dll` | Native library shipped by the KeyGuard team. Place it next to the executable. |
+| Access to an MAA instance | This sample uses the **EUS2 shared test endpoint**: `Replace with your own.` |
+
+> 💡 **Tip** If the DLL depends on VC++ runtimes make sure they’re installed (see [Troubleshooting](#7️⃣ troubleshooting)).
+
+---
+
+## 💻 Enable KeyGuard VBS
+
+Only applicable on **Windows**. These commands must be run **once per machine** with Administrator rights.
+
+```powershell
+reg add "HKLM\SYSTEM\CurrentControlSet\Control\DeviceGuard" /v "EnableVirtualizationBasedSecurity" /t REG_DWORD /d 1 /f
+reg add "HKLM\SYSTEM\CurrentControlSet\Control\DeviceGuard" /v "RequirePlatformSecurityFeatures"       /t REG_DWORD /d 1 /f
+reg add "HKLM\SYSTEM\CurrentControlSet\Control\DeviceGuard" /v "Locked"                                /t REG_DWORD /d 0 /f
+reg add "HKLM\SYSTEM\CurrentControlSet\Control\DeviceGuard\Scenarios\HypervisorEnforcedCodeIntegrity" /v "Enabled" /t REG_DWORD /d 1 /f
+reg add "HKLM\SYSTEM\CurrentControlSet\Control\DeviceGuard\Scenarios\HypervisorEnforcedCodeIntegrity" /v "Locked"  /t REG_DWORD /d 0 /f
+```
+
+> **Reboot** afterwards and verify the **Windows Security Service** is running:  
+> ```powershell
+> sc query SecurityHealthService
+> ```  
+> The state must be **RUNNING** every time you launch the PoC.
+
+---
+
+## 2️⃣ Build
+
+```bash
+# Clone / copy the sources
+git clone https://.../KeyGuardDemo.git
+cd KeyGuardDemo
+
+# Restore packages from the **internal NuGet feed**
+dotnet restore    # make sure your *NuGet.Config* points to the https://msazure.visualstudio.com/One/_artifacts/feed/Official/NuGet/Microsoft.Azure.Security.KeyGuardAttestation feed
```

Comment:
This is going to be a pain to install. Does the MAA team agree to put this on nuget?